torcharrow/test/integration/test_criteo.py

```diff
@@ -0,0 +1,217 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+import os
+import random
+import unittest
+from typing import List
+
+import numpy as np
+import numpy.testing
+import torch
+import torcharrow as ta
+import torcharrow.dtypes as dt
+import torcharrow.pytorch as tap
+from torcharrow.functional import functional
+
+try:
+    import pyarrow.parquet as pq
+
+    HAS_PYARROW_PARQUET = True
+except ImportError:
+    HAS_PYARROW_PARQUET = False
+skipIfNoPyArrowParquet = unittest.skipIf(not HAS_PYARROW_PARQUET, "no PyArrow Parquet")
+
+# Based on https://github.com/facebookresearch/torchrec/blob/9a8a4ad631bbccd7cd8166b7e6d7607e2560d2bd/torchrec/datasets/criteo.py#L37-L46
+INT_FEATURE_COUNT = 13
+CAT_FEATURE_COUNT = 26
+DEFAULT_LABEL_NAME = "label"
+DEFAULT_INT_NAMES: List[str] = [f"int_{idx}" for idx in range(INT_FEATURE_COUNT)]
+DEFAULT_CAT_NAMES: List[str] = [f"cat_{idx}" for idx in range(CAT_FEATURE_COUNT)]
+
+DTYPE = dt.Struct(
+    [
+        dt.Field(DEFAULT_LABEL_NAME, dt.int8),
+        dt.Field(
+            "dense_features",
+            dt.Struct(
+                [
+                    dt.Field(int_name, dt.Int32(nullable=True))
+                    for int_name in DEFAULT_INT_NAMES
+                ]
+            ),
+        ),
+        dt.Field(
+            "sparse_features",
+            dt.Struct(
+                [
+                    dt.Field(cat_name, dt.Int32(nullable=True))
+                    for cat_name in DEFAULT_CAT_NAMES
+                ]
+            ),
+        ),
+    ]
+)
+
+# TODO: allow to use to_tensor with Callable
+# TODO: implement conversion in native C++
+class _DenseConversion(tap.ITorchConversion):
+    def to_tensor(self, df: ta.IDataFrame):
+        # Default to_tensor, each field is a Tensor
+        tensors = df.to_tensor()
+
+        # Stack them into BatchSize * NumFields Tensor
+        return torch.cat(
+            [column_tensor.values.unsqueeze(0).T for column_tensor in tensors], dim=1
+        )
```

Comment:
Is each `column_tensor` or `column_tensor.values` a real Tensor?
nit: Would `torch.cat([column_tensor.values.unsqueeze(0) for column_tensor in tensors]).T` be easier to understand?

Reply:
> Is each column_tensor or column_tensor.values a real Tensor?

`column_tensor.values ` is the real Tensor -- i think i really need to hack around the nullability issue :)